### ISSUE TITLE
CreateObject component: The order of the inputs depends on the sequence we select/deselect them

### ISSUE THREAD

**Issue by epignatelli** [BHoM/BHoM_UI#156]
#### How to replicate:

1. Checkout this branch https://github.com/BHoM/BHoM_UI/pull/152 and compile the code accordingly
1. Open Grasshopper
1. Create a `CreateObject` component
1. Select the `oM.Structure.Elements.Bar` constructor by object initializer
1. Right click, and use the Add/Remove Inputs functionality
1. Remove the `StartNode` property
1. Re-add the `StartNode` property
1. Note that it appears at the bottom now.

#### Expected behaviour:

I think we can use the index of the element in the properties list of the object, in conjunction with the `AddInput` method to place the input correctly. 
```pseudo
if IndexOfProperty < InputParams.Length 
  Add it to the right place (Grasshopper should take care of shifting the rest of them)
  OnBHoMUpdates()
else
   Add it to the end of the list
```

**Comment by adecler** [comment]
I was aware of it but didn't see that as a problem. It can even have its applications as it gives more control to the user. I guess the line between feature and bug can sometimes be thin.

Happy to go either way.

**Comment by al-fisher** [comment]
Hmmm... I think unless we give the user complete control - i.e. can easily inject or drag properties up and down to curate their own list (which I think is quite a way beyond scope of current UI incarnation) - then maybe more intuitive to have constant list order actually.

And also - this wIll help users across shared scripts also. 

Thinking nowI wonder if the parameter order on component and list order in Add/Remove should be the same as the argument order in the method signature in the original create method selection also. With the additional (FragmentSet etc.) always as the end. Nice to have this clear consistency across all - so method will always look familiar

**Comment by adecler** [comment]
This will be fixed once the PR on the UI refactoring is merged: https://github.com/BHoM/BHoM_UI/pull/297
